crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -1684,15 +1684,27 @@ pub(super) fn report_implicit_return_type(
     expected_ty: Type,
     has_empty_body: bool,
     enclosing_class_of_method: Option<ClassLiteral>,
+    no_return: bool,
 ) {
     let Some(builder) = context.report_lint(&INVALID_RETURN_TYPE, range) else {
         return;
     };
     let db = context.db();
-    let mut diagnostic = builder.into_diagnostic(format_args!(
-        "Function can implicitly return `None`, which is not assignable to return type `{}`",
-        expected_ty.display(db)
-    ));
+
+    // If no return statement is defined in the function, then the function always return `None`
+    let mut diagnostic = if no_return {
+        let mut diag = builder.into_diagnostic(format_args!(
+            "Function always implicitly return `None`, which is not assignable to return type `{}`",
+            expected_ty.display(db),
+        ));
+        diag.info("Consider changing your return annotation to `-> None`");
```

Comment:
Personally I'm not clear why we would assume it's the annotation that's wrong, rather than a `return` statement having been forgotten in the function body. @AlexWaygood I think this info hint was your suggestion; any comment?

(Also, minor wording nit, I think we should avoid describing the annotation as "your annotation" -- we don't know if the person who wrote it is the same person reading the diagnostic.)

```suggestion
        diag.info("Consider changing the return annotation to `-> None` or adding a `return` statement");
```

Reply:
> Personally I'm not clear why we would assume it's the annotation that's wrong, rather than a `return` statement having been forgotten in the function body. @AlexWaygood I think this info hint was your suggestion; any comment?

agree that it makes sense to give both hints, yes!